actions/generate-schedule/scheduler/scheduler.go

```diff
@@ -0,0 +1,104 @@
+package scheduler
+
+import (
+	"fmt"
+	"github.com/spinnaker/rotation-scheduler/actions/generate-schedule/users"
+	"github.com/spinnaker/rotation-scheduler/proto/schedule"
+	"time"
+)
+
+const (
+	DateFormat = "Mon 02 Jan 2006"
+)
+
+// Scheduler creates or extends a new rotation schedule.
+type Scheduler struct {
+	userSource        users.Source
+	shiftDurationDays int
+}
```

Comment:
Personally I would be inclined to include this data in the `Schedule` message, but not a huge deal and I guess we can decide if that makes sense that later...